### PR TITLE
remove hard-coded, statically allocated array sizes for tmpT and tmpZ

### PR DESCRIPTION
**Pull request by bartnijssen**
@yixinmao and @jhamman can you review this - seems like a lot of work for a style change. I'm fine if we don't use this.

@yixinmao - can you run my branch with your test cases to make sure there is no bug in how I allocate the memory?

### CODE REVIEW

**Comment by bartnijssen** [comment]
Hold off on testing - it is core dumping so let me fix it first on my end.

**Comment by bartnijssen** [comment]
My travis build passes: https://travis-ci.org/bartnijssen/VIC/builds/132987645

So ready for testing. Before you merge let's decide if we want this. If you do merge, please squash my two commits

File: vic/drivers/shared_all/src/compute_derived_state_vars.c [hunk 1]
```diff
@@ -46,13 +46,20 @@ compute_derived_state_vars(all_vars_struct *all_vars,
     size_t                     veg;
     size_t                     lidx;
     size_t                     band;
+    size_t                     tmpTshape[] = {
+        MAX_LAYERS, MAX_NODES,
+        MAX_FROST_AREAS + 1
+    };
+    size_t                     tmpZshape[] = {
+        MAX_LAYERS, MAX_NODES
+    };
```
Comment (jhamman) [inline vic/drivers/shared_all/src/compute_derived_state_vars.c:56]: did uncrustify do this indentation?

**Comment by yixinmao** [comment]
@bartnijssen @jhamman : I've tested Bart's branch and it works fine on my test cases. I personally think the changes are good, but let you know what you think.

**Comment by jhamman** [comment]
This looks fine to me.  I'll suggest that either @bartnijssen (here) or @yixinmao (after the merge) should add a comment near where `tmpTshape` and `tmpZshape` are defined that explains why we're choosing maximum bounds.  I think a follow on to this PR would be to change `MAX_LAYERS --> options.nlayers` and `MAX_NODES --> options.nodes` since that is the actual size of the useful array.

File: vic/drivers/shared_all/src/compute_derived_state_vars.c (hunk 1 above)
Comment (bartnijssen) [inline vic/drivers/shared_all/src/compute_derived_state_vars.c:56]: Indeed - I think it's ugly - but uncrustify is going to change this every time, so I'll just leave it.

**Comment by bartnijssen** [comment]
Good points - I'll open an issue and assign it to @yixinmao who can then determine whether to change this or not.

**Comment by bartnijssen** [comment]
I'll open another issue to generalize and collect all these n-dimensional memory allocations and treat them all the same way

**Comment by bartnijssen** [comment]
@yixinmao : Once this passes the travis build (only added a comment so should be OK) - this should be ready to merge into your branch, which will then update the PR to the mothership

**Comment by bartnijssen** [comment]
BTW - please squash the commits when you merge

**Comment by bartnijssen** [comment]
@yixinmao : Can you make sure you also test this with `valgrind` before you merge this. Since this involves dynamic allocation (and freeing) it is a natural entry for memory leaks.

**Comment by yixinmao** [comment]
@bartnijssen: Do you mean to run with `valgrind` and see if there is memory error?

**Pull request merged**